app/ota_client_stub.py

```diff
@@ -581,31 +532,46 @@ async def status(self, _: Optional[v2.StatusRequest] = None) -> v2.StatusRespons
         """
         cur_time = time.time()
         if cur_time - self._last_status_query > server_cfg.STATUS_UPDATE_INTERVAL:
-            # for status API, query all directly connected subecus
-            tracked_ecu = set(
-                [ecu["ecu_id"] for ecu in self._ecu_info.get_secondary_ecus()]
-            )
             async with self._status_pulling_lock:
                 self._last_status_query = cur_time
-                # prepare subecu status
-                _response = await self._query_subecu_status(tracked_ecu)
-
-                # prepare my ecu status
-                ecu_id = self._ecu_info.get_ecu_id()  # my ecu id
-                ecu = _response.ecu.add()
-                if status := self._ota_client.status():
-                    ecu.CopyFrom(status)
+
+                # get subecus' status
+                subecus_resp = v2.StatusResponse()
+                coros: List[Coroutine] = []
+                for subecu_id, subecu_addr in self.subecus_dict.items():
+                    coros.append(
+                        self._ota_client_call.status_call(
+                            subecu_id,
+                            subecu_addr,
+                            timeout=server_cfg.QUERYING_SUBECU_STATUS_TIMEOUT,
+                        )
+                    )
+                subecu_resp: v2.StatusResponse
+                for subecu_resp in await asyncio.gather(*coros):
+                    # gather the subecu and its child ecus status
+                    for _ecu_resp in subecu_resp.ecu:  # type: ignore
+                        _ecu = subecus_resp.ecu.add()
+                        _ecu.CopyFrom(_ecu_resp)
+
+                # prepare final response
+                resp = v2.StatusResponse()
+                # NOTE: only contains directly connected subecus!
```

Comment:
`NOTE: only contains directly connected subecus!`
available_ecu_ids means indicates all ecus for OTA target. (not only directly connected sub ecus)
See https://github.com/tier4/ota-client/blob/main/docs/README.md
I think implementation is OK, but the comment misreads.

Reply:
@obi-t4 I would like to confirm one thing, only main ecu has `available_ecu_ids` in its `ecu_info.yaml`, is that correct?

Reply:
Correct, but this is an operation rule.
But we don't need to check if `I'm main ecu`.